hs_core/tests/api/rest/test_resource_flags.py

```diff
@@ -0,0 +1,67 @@
+from rest_framework import status
+
+from hs_core.hydroshare import resource
+
+from .base import HSRESTTestCase
+
+
+class TestPublicResourceFlagsEndpoint(HSRESTTestCase):
+    def setUp(self):
+        super(TestPublicResourceFlagsEndpoint, self).setUp()
+
+        self.rtype = 'GenericResource'
+        self.title = 'My Test resource'
+        res = resource.create_resource(self.rtype,
+                                       self.user,
+                                       self.title)
+
+        self.pid = res.short_id
+        self.resources_to_delete.append(self.pid)
+
+    def test_set_resource_flag_make_public(self):
```

Comment:
This test should be failing as it is not possible to set its flag to 'public' when this resource does not have any resource file and missing required metadata (abstract and keyword).